rules/library.bzl

```diff
@@ -536,6 +540,46 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
                 opts = repr(v),
             ))
 
+    # Generate code for `.intentdefinition` source files and forward the original plist as data
+    for intent in intent_sources:
+        intent_name = "%s_%s_gen" % (name, intent)
+        testonly = kwargs.get("testonly", False)
+
+        # Avoid swift_intent_library and objc_intent_library because they wrap the generated code in a new module
+        #
+        # If other Swift sources are present, generate Swift intent code, otherwise use Obj-C.
+        # This mimics the behavior for INTENTS_CODEGEN_LANGUAGE="automatic" in Xcode.
+        if len(swift_sources) > 0:
+            apple_intent_library(
+                name = intent_name,
+                src = intent,
+                language = "Swift",
+                tags = _MANUAL,
+                testonly = testonly,
+            )
+
+            swift_sources.append(intent_name)
+            data.append(intent)
+
+        else:
+            intent_headers = "%s_hdrs" % intent_name
```

Comment:
Is it a logical error if they add both `objc` and `swift` here - maybe consider a `fail`

Reply:
I don't think it should be possible since we're inferring from the given `srcs` and it's not an explicit user choice.  A Swift-only or mixed language target will hit the true block and all others would generate Obj-C in the false block.